lib/pbench/server/api/resources/query_apis/datasets_detail.py

```diff
@@ -0,0 +1,266 @@
+
+        "name" is the name of a Pbench agent dataset (tarball).
+
+        "start" and "end" are time strings representing a set of Elasticsearch
+        run document indices in which the dataset will be found.
+
+        The Elasticsearch URL string is also dependent upon the configured
+        "prefix". This is a Pbench artifact that allows multiple Pbench
+        instances to share a single Elasticsearch server, by segregating
+        their data into unique namespaces. Each Pbench index name is
+        qualified by the index_prefix of the Pbench server that originally
+        indexed the data. The prefix comes from the pbench-server.cfg
+        file for the server instance. For example, production, "staging",
+        and development server instances might all share one Elasticsearch
+        cluster by choosing unique prefixes to generate unique index
+        names.
+
+        Required headers include
+
+        Authorization:  Pbench authorization token authorized to access
+                        "user"'s data. E.g., "user", "admin" or a user with
+                        whom the dataset has been shared.
+        Content-Type:   application/json
+        Accept:         application/json
+
+        NOTE: This is the format currently constructed by the Pbench
+        dashboard `src/model/dashboard.js` fetchResults effect. The
+        "runMetadata" property is a union of the run document's "@metadata
+        and "run" sub-documents, while "hostTools" is the run document's
+        "host_tools_info" nested array.
+
+        TODO: We probably need to return all *unowned* runs when called
+        without a session token or user parameter. We need to define precisely
+        how this should work.
+
+        [
+            {
+                "runMetadata": {
+                    "file-name": "/pbench/archive/fs-version-001/dhcp31-187.perf.lab.eng.bos.redhat.com/fio_rhel8_kvm_perf43_preallocfull_nvme_run4_iothread_isolcpus_2020.04.29T12.49.13.tar.xz",
+                    "file-size": 216319392,
+                    "md5": "12fb1e952fd826727810868c9327254f",
+                    [...]
+                },
+                "hostTools": [
+                    {
+                        "hostname": "dhcp31-187",
+                        "tools": {
+                            "iostat": "--interval=3",
+                            [...]
+                        }
+                    }
+                ]
+            }
+        ]
+        """
+        json_data = request.get_json(silent=True)
+        if not json_data:
+            self.logger.info("Invalid JSON object. Query: {}", request.url)
+            abort(400, message="Invalid request payload")
+
+        try:
+            user = json_data["user"]
+            name = json_data["name"]
+            start_arg = json_data["start"]
+            end_arg = json_data["end"]
+        except KeyError:
+            keys = [k for k in ("user", "name", "start", "end") if k not in json_data]
```

Comment:
Since we are going through the effort to generate the list of keys as a list, only to immediately format them as a string, perhaps we can do that once and just reference the string?